src/plugins/data/common/index_patterns/index_patterns/types.ts

```diff
@@ -33,3 +33,16 @@ export interface TypeMeta {
   aggs?: Record<string, AggregationRestrictions>;
   [key: string]: any;
 }
+
+export interface IndexPatternSpec {
+  id: string;
+  type: string;
+  source: {
+    title: string;
+    timeFieldName: string;
+    sourceFilters: string;
```

Comment:
i think sourceFilters are unused, they are a private variable which is not used at all, lets skip it from the spec

Reply:
Its used by `searchSource` via `getSourceFiltering`

Reply:
but are they ever set ?

Reply:
they're set via `_.assign`